### PR TITLE
Ch65/create variant and segment models

### PR DESCRIPTION
**Pull request by AlineRibeiro**

### CODE REVIEW

File: db/migrate/20210615010628_create_variants.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class CreateVariants < ActiveRecord::Migration[6.1]
+  def change
+    create_table :variants do |t|
+      t.references :video, null: false, foreign_key: true
+      t.string :public_id
+      t.integer :height
+      t.integer :width
+      t.integer :bitrate
```
Comment (guialbuk) [inline db/migrate/20210615010628_create_variants.rb]: All these columns must be `null: false`

File: db/migrate/20210615014337_create_segments.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class CreateSegments < ActiveRecord::Migration[6.1]
+  def change
+    create_table :segments do |t|
+      t.references :variant, null: false, foreign_key: true
+      t.integer :position
+      t.float :duration
```
Comment (guialbuk) [inline db/migrate/20210615014337_create_segments.rb]: These columns must be `null: false` as well.

File: spec/models/segment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  it ' has a valid factory' do
+    expect(segment.validate!).to eq(true)
+  end
+
+  it { is_expected.to validate_numericality_of(:position).is_greater_than(0) }
+
+  it { is_expected.to validate_numericality_of(:duration).is_greater_than(0) }
+
+  it { is_expected.to validate_uniqueness_of(:position).scoped_to(:variant_id) }
```
Comment (guialbuk) [inline spec/models/segment_spec.rb:16]: 👍

Comment (AlineRibeiro) [reply spec/models/segment_spec.rb:16]: Thank you!

**Pull request merged**